src/targets/gpu/kernels/include/migraphx/kernels/debug.hpp

```diff
@@ -178,5 +178,9 @@ MIGRAPHX_HIP_NORETURN inline __host__ __device__ void assert_fail(const source_l
 #define MIGRAPHX_WARN(...)
 #endif
 
+#define MIGRAPHX_STATIC_ASSERT_FOR(...) \
+    static_assert(__VA_ARGS__);         \
+    if constexpr(__VA_ARGS__)
```

Comment:
Two questions about this: 
What is the reason for using FOR in the name? 
Why isn't static_assert sufficient on its own?

Reply:
>  What is the reason for using FOR in the name?

I am not sure if this is the best name, but it means we are applying a `static_assert` for the block.

>  Why isn't static_assert sufficient on its own?

When the `static_assert` is `false` the compiler will still continue compiling, but this macro will prevent the block from compiling if the assert is wrong. The problem is that when `max_stride_iterations` grows too large, the `static_assert` will hit, but it will contine to compile, which is very slow. So this will allow to skip the loop unroll if the `static_assert` is `false`.